webapp/src/main/webapp/js/extension.js

```diff
@@ -28,10 +28,58 @@
       color: '#CB4B32',
     }
   ];
+  const supportedDocumentsExtensionOptions = [
+    {
+      provider: 'onlyoffice',
+      extension: '.docx',
+      mimeType: 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+      view: true,
+      edit: true
+    },
+    {
+      provider: 'onlyoffice',
+      extension: '.docxf',
+      mimeType: 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+      view: true,
+      edit: true
+    },
+    {
+      provider: 'onlyoffice',
+      extension: '.pptx',
+      mimeType: 'application/vnd.openxmlformats-officedocument.presentationml.presentation',
+      view: true,
+      edit: true
+    },
+    {
+      provider: 'onlyoffice',
+      extension: '.xlsx',
+      mimeType: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+      view: true,
+      edit: true
+    },
+    {
+      provider: 'onlyoffice',
+      extension: '.oform',
+      mimeType: 'application/vnd.openxmlformats-officedocument.wordprocessingml.document.form',
+      view: true,
+      edit: true
+    },
+    {
+      provider: 'onlyoffice',
+      extension: '.pdf',
+      mimeType: 'application/pdf',
+      view: true,
+      edit: false
+    },
+  ];
+
   const lang = eXo.env.portal.language || 'en';
   const url = `${eXo.env.portal.context}/${eXo.env.portal.rest}/i18n/bundle/locale.navigation.portal.global-${lang}.json`;
 
   exoi18n.loadLanguageAsync(lang, url).then(i18n => new Vue({i18n}));
   newDocumentTypeExtensionOptions.forEach(extension => extensionRegistry.registerExtension('attachment', 'new-document-action', extension));
   document.dispatchEvent(new CustomEvent('attachment-new-document-action-updated'));
+
+  supportedDocumentsExtensionOptions.forEach(extension => extensionRegistry.registerExtension('documents', 'supported-document-types', extension));
+  document.dispatchEvent(new CustomEvent('documents-supported-document-types-updated'));
```

Comment:
IMO, it would be better to add an `extensionRegistry.loadExtensions('documents-supported-mimetype', 'edit')` + `extensionRegistry.registerExtension(...)` in OO

Reply:
each extension contains two props view and edit to tell if each supported for view and edit or only edit, so the extension will be more general in term of use and (also has mime-type and extension props) .
https://github.com/exoplatform/onlyoffice/pull/167/files#diff-8713d388b9ba6f041081ad2c03b539adc3bbf9b45b243b06772e3d78b7f684bdR71